sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ResolveWriteToStream.scala

```diff
@@ -89,11 +89,12 @@ object ResolveWriteToStream extends Rule[LogicalPlan] with SQLConfHelper {
             s"""SparkSession.conf.set("${SQLConf.CHECKPOINT_LOCATION.key}", ...)""")
       }
     }
+    val fileManager = CheckpointFileManager.create(new Path(checkpointLocation), s.hadoopConf)
+
     // If offsets have already been created, we trying to resume a query.
     if (!s.recoverFromCheckpointLocation) {
       val checkpointPath = new Path(checkpointLocation, "offsets")
-      val fs = checkpointPath.getFileSystem(s.hadoopConf)
-      if (fs.exists(checkpointPath)) {
+      if (fileManager.exists(checkpointPath)) {
```

Comment:
@viirya @HeartSaVioR do you know why `FileSystemBasedCheckpointFileManager.exists` calls `fs.getFileStatus`? It's inconsistent with the old code.

Reply:
I'm not sure about the reason. It seems to be `getFileStatus` since `FileSystemBasedCheckpointFileManager` was added. And looks like `exists` basically calls `getFileStatus`.

Reply:
Could you please refer the old code? The initial implementation of FileSystemBasedCheckpointFileManager calls getFileStatus in exists.

https://github.com/apache/spark/commit/cbb41a0c5b01579c85f06ef42cc0585fbef216c5

Reply:
The old code is just one line above here.

Reply:
It is "effectively" consistent as exists will call getFileStatus anyway, but I see that FileSystemBasedCheckpointFileManager.exists can just call `fs.exists` rather than calling `fs.getFileStatus`. It's even inconsistent with other methods in FileSystemBasedCheckpointFileManager - other methods call `fs.exists`.

Probably the better fix would be changing FileSystemBasedCheckpointFileManager.exists to call fs.exists.

Reply:
https://github.com/apache/spark/pull/39936